storage/series/buffer.go

```diff
@@ -680,9 +667,7 @@ func (b *dbBufferBucket) resetEncoders() {
 	for i := range b.encoders {
 		// Register when this bucket resets we close the encoder
 		encoder := b.encoders[i].encoder
-		if b.ctx != nil {
-			b.ctx.RegisterCloser(encoder)
-		}
+		encoder.Close()
```

Comment:
do you need to take ctx as an arg in these methods?

Reply:
When we call `encoder.Stream()` we take a copy of the bytes and pass back a new Segment wrapping them, so don't need to involve ctx at all no, once we have the write lock on the series (which we have here) its safe to close.